### PR TITLE
Add support for launch volumes

### LINKED ISSUE #363: Add support for launch volumes in Self managed cluster
**What would you like to be added**:
Add support for launch volumes in Self managed cluster. https://docs.oracle.com/en-us/iaas/api/#/en/iaas/20160918/datatypes/LaunchAttachVolumeDetails

**Why is this needed**:

This will help customers attach block volumes during launch.

### PR DESCRIPTION
**Pull request by shyamradhakrishnan**
**What this PR does / why we need it**:

Add support for launch volumes

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #363

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -778,8 +778,8 @@ sigs.k8s.io/controller-runtime v0.16.5 h1:yr1cEJbX08xsTW6XEIzT13KHHmIyX8Umvme2cU
 sigs.k8s.io/controller-runtime v0.16.5/go.mod h1:j7bialYoSn142nv9sCOJmQgDXQXxnroFU4VnX/brVJ0=
 sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd h1:EDPBXCAspyGV4jQlpZSudPeMmr1bNJefnuqLsRAsHZo=
 sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd/go.mod h1:B8JuhiUyNFVKdsE8h686QcCxMaH6HrOAZj4vswFpcB0=
-sigs.k8s.io/kind v0.20.0 h1:f0sc3v9mQbGnjBUaqSFST1dwIuiikKVGgoTwpoP33a8=
-sigs.k8s.io/kind v0.20.0/go.mod h1:aBlbxg08cauDgZ612shr017/rZwqd7AS563FvpWKPVs=
+sigs.k8s.io/kind v0.21.0 h1:QgkVrW35dMXNLkWlUkq2uFQNQbPLr0Z6RgRH5P/NzZU=
+sigs.k8s.io/kind v0.21.0/go.mod h1:aBlbxg08cauDgZ612shr017/rZwqd7AS563FvpWKPVs=
```
Comment (joekr) [inline go.sum:782]: Is this just adding support for launch volumes via kind?

Comment (shyamradhakrishnan) [reply go.sum:782]: Hey Joe, did not get you, we are adding a new parameter in machine spec called launch volumes and send them to compute API right? Sorry I may be missing something here. The kind change is to update the kind version, as kind 0.20 was not running fine.

**Comment by shyamradhakrishnan** [comment]
unit test
```
$ make test
GOBIN=/home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci/hack/tools/bin ./scripts/go_install.sh sigs.k8s.io/controller-runtime/tools/setup-envtest setup-envtest v0.0.0-20230131074648-f5014c077fc3
kube-builder assets: /home/ubuntu/.local/share/kubebuilder-envtest/k8s/1.24.2-linux-amd64
KUBEBUILDER_ASSETS="/home/ubuntu/.local/share/kubebuilder-envtest/k8s/1.24.2-linux-amd64" go test -coverprofile=coverage.out ./...
?   	github.com/oracle/cluster-api-provider-oci	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/config	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/metrics	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/api/v1beta1	51.344s	coverage: 22.9% of statements
ok  	github.com/oracle/cluster-api-provider-oci/api/v1beta2	0.035s	coverage: 15.0% of statements
ok  	github.com/oracle/cluster-api-provider-oci/cloud/ociutil	0.012s	coverage: 19.6% of statements
?   	github.com/oracle/cluster-api-provider-oci/cloud/scope/mocks	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/base	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/base/mock_base	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/compute	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/compute/mock_compute	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/computemanagement	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/computemanagement/mock_computemanagement	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/containerengine	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/containerengine/mock_containerengine	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/identity	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/identity/mock_identity	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/loadbalancer	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/loadbalancer/mock_lb	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer/mock_nlb	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/vcn	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/vcn/mock_vcn	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/feature	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/version	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/cloud/scope	230.727s	coverage: 76.1% of statements
ok  	github.com/oracle/cluster-api-provider-oci/cloud/util	0.160s	coverage: 61.1% of statements
ok  	github.com/oracle/cluster-api-provider-oci/controllers	26.983s	coverage: 58.7% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/api/v1beta1	13.649s	coverage: 15.3% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/api/v1beta2	0.016s	coverage: 4.8% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/controllers	0.259s	coverage: 56.3% of statements
go tool cover -func=coverage.out -o coverage.txt
go tool cover -html=coverage.out -o coverage.html
$
```

**Comment by shyamradhakrishnan** [comment]
e2e
```

Ran 7 of 28 Specs in 2828.523 seconds
SUCCESS! -- 7 Passed | 0 Failed | 0 Pending | 21 Skipped
```

File: go.sum (hunk 1 above)
Comment (joekr) [reply go.sum:782]: sorry when I looked at this PR this and the go.mod files were the only thing I saw changed. That is why I was confused 🤣

I'll look this over today

File: cloud/scope/clients.go [hunk 2]
```diff
@@ -192,19 +192,6 @@ func (c *ClientProvider) createVcnClient(region string, ociAuthConfigProvider co
...
-			err = c.setCerts(client)
-			if err != nil {
-				logger.Error(err, "unable to create OCI VCN Client")
-				return nil, err
-			}
-		} else {
-			return nil, errors.New("The VCN Client dispatcher is not of http.Client type. Can not patch the tls config.")
-		}
-
-	}
```
Comment (joekr) [inline cloud/scope/clients.go:206]: And this is removed since we can now set the certs via an ENV?

**Review by joekr — APPROVED** [review]
Not saying we need this now, but should an e2e test launch a machine with a Launch volume set?

File: cloud/scope/clients.go (hunk 2 above)
Comment (shyamradhakrishnan) [reply cloud/scope/clients.go:206]: yep https://github.com/oracle/oci-go-sdk/blob/master/common/tls_config_provider.go#L17

Comment (shyamradhakrishnan) [reply cloud/scope/clients.go:206]: Also, we have added volumes in the e2e template? We are missing validation there though. I can add it as separate PR.

**Pull request merged**